### ISSUE TITLE
OpenIAB Xcode Build Error

### ISSUE THREAD

**Issue by emakbaba** [onepf/OpenIAB#407]
Hello,

Im using Simple IAP and OpenIAB system.

Not problem Android system but i tried Xcode for IOS  I have an error.

I added StoreKit framework.

Please see attach file.

![xcode2](https://cloud.githubusercontent.com/assets/6122814/5297535/788af1d2-7bb8-11e4-8015-19b5eb8fef69.png)

How can i fix this?

Thanks.

**Comment by fromparis** [comment]
same problem, first 

in this file :
game/Libraries/Plugins/iOS/AppStoreDelegate.mm

'release' is unavailable: not available in automatic reference counting mode

so i change the apple LLVM 6.0 automatic (...) to NO then i get the same problem..

**Comment by GrimReio** [comment]
Hello!
Couldn't reproduce this issue. When I generate XCode project in Unity, automatic ref counting is off by default.
What versions of XCode and iOS SDK do you have?

**Comment by fromparis** [comment]
hi , again, 

well i'm using xcode 6.2, i'm using unity 5, i'v tested witch unity 4.6 it works fine, but in unity 5
it's a no go, i'm using  openiab-plugin-0.9.7.2.unitypackage

![screen shot 2015-01-28 at 19 13 23](https://cloud.githubusercontent.com/assets/10722282/5943734/d0f5e46e-a721-11e4-9a32-e369d244ba2a.png)

**Comment by fromparis** [comment]
and when i change the automatic to no
![screen shot 2015-01-28 at 19 25 27](https://cloud.githubusercontent.com/assets/10722282/5943971/922fdf58-a723-11e4-92e4-25fa9a249db1.png)

**Comment by fromparis** [comment]
i get 
![screen shot 2015-01-28 at 19 25 51](https://cloud.githubusercontent.com/assets/10722282/5943980/af6ed59c-a723-11e4-8ac5-a8712091ec4d.png)

**Comment by GrimReio** [comment]
I'm sorry, I can't get access to Unity 5 right now. I bet they changed something in the build process. I'll fix it when Unity 5 will be available.

**Comment by ray-garner** [comment]
We need this please.  Unity 5 was released last week.

**Comment by xyzDave** [comment]
Had the same issue - I believe you can fix it by replacing the dealloc and requestSKUs functions in AppStoreDelegate.mm as below.  My ObjectiveC isn't great, so someone let me know if this is wrong, but seems to work for me :)

```
- (void)dealloc
{
    [[SKPaymentQueue defaultQueue] removeTransactionObserver:self];
//    [m_skuMap release];
//    [m_skus release];
    m_skus = nil;
    m_skuMap = nil;
//    [super dealloc];
}
```
```
// Setup

- (void)requestSKUs:(NSSet*)skus
{
//    m_skus = [skus retain];
    m_skus = skus;
    SKProductsRequest *request = [[SKProductsRequest alloc] initWithProductIdentifiers:skus];
	request.delegate = self;
	[request start];
}
```

**Comment by ray-garner** [comment]
Yes I did the same thing but hopefully we can get an official fix?   This kills your Unity Cloud build

**Comment by xyzDave** [comment]
Did you change the file in Assets/Plugins/iOS/AppstoreDelegate.mm or the one copied to the Xcode build folder?
Official fix would be very nice - fingers crossed it's done for release soon as lots of people will be having this issue.

**Comment by ray-garner** [comment]
Guess I should have done it in ios plugins... hehe.  Well that would fix that.  @xyzDave

**Comment by tyomklz** [comment]
Hello there. Exactly the same problem. Is it fixed yet? It looks like you may just comment some lines here and there and then it'll build okay, but doesn't it also means that **mapSku** is going to stay unmapped then? And I basically mean this:
```C#
OpenIAB.mapSku(SKU_ITEM, OpenIAB_iOS.STORE, "SKU_NAME");
```
Though I barely understand what exactly is going on there, just guesses. What should I do?

**Comment by waqasjatala** [comment]
Hello every body , i have the above issue , my unity version is 5.5.1 , xcode version is 8.2.1, ios sdk version 10.2, mac version is 10.12. Thanks in advance.
<img width="1167" alt="d0f5e46e-a721-11e4-9a32-e369d244ba2a" src="https://user-images.githubusercontent.com/29325055/27000159-6fc7c18e-4dc4-11e7-9cd7-ee15038f3ba1.png">
